service/src/main/java/com/epam/esm/service/mapper/impl/CertificateMapper.java

```diff
@@ -0,0 +1,55 @@
+package com.epam.esm.service.mapper.impl;
+
+import com.epam.esm.dao.entity.Certificate;
+import com.epam.esm.dao.entity.Tag;
+import com.epam.esm.service.dto.CertificateDto;
+import com.epam.esm.service.dto.TagDto;
+import com.epam.esm.service.mapper.Mapper;
+import org.springframework.stereotype.Component;
+
+import java.util.Collections;
+import java.util.List;
+import java.util.Set;
+import java.util.stream.Collectors;
+
+@Component
+public class CertificateMapper implements Mapper<Certificate, CertificateDto> {
+  @Override
+  public CertificateDto mapToDto(Certificate entity, boolean mapNestedEntities) {
```

Comment:
@Exqzore Could you use for this action MapStruct https://mapstruct.org/ or ModelMapper http://modelmapper.org/

Reply:
I need my implementation specifically because I can flexibly control this